### PR TITLE
New PR and Issue templates

### PR DESCRIPTION
**Pull request by aimurphy**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your PR.

## Purpose

Update the PR template to reflect our current practices.

Also updated the issue template.

Took some inspiration from Datadog's bare bones PR template: https://github.com/DataDog/documentation/blob/master/.github/PULL_REQUEST_TEMPLATE.md 

## Merge requirements

- [x] Successful preview build.
- [x] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

### CODE REVIEW

**Review by SushrutHarness — APPROVED** [review]
LGTM

**Review by brian-f-harness — DISMISSED** [review]
LGTM

**Review by douglas-j-bothwell — DISMISSED** [review]
LGTM. I like the checklist!

**Review by ravilach — DISMISSED** [review]

**Comment by ravilach** [comment]
I would say to limit the checklist for folks in the public who might not know who a SME is. Can separate Harness vs Non-Harness.

**Comment by aimurphy** [comment]
> I would say to limit the checklist for folks in the public who might not know who a SME is. Can separate Harness vs Non-Harness.

good point. changed it to be shorter and just say generic "code owner review"

**Comment by krishi0408** [comment]
LGTM

**Review by krishi0408 — APPROVED** [review]

**Review by roshnisarangadharan — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,19 +1,22 @@
-# Harness Developer Pull Request
-Thanks for helping us make the Developer Hub better. The PR will be looked at
-by the CODEOWNERS. 
+Thanks for contributing to the Harness Developer Hub! Our code owners will review your PR.
```
Comment (SmritiSatya) [inline .github/pull_request_template.md:1]: Thank you for contributing to the Harness Developer Hub! Our code owners will review your pull request (PR).

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 
-## What Type of PR is This?
+## Purpose
 
-- [ ] Issue
-- [ ] Feature
-- [ ] Maintenance/Chore
+What is the motivation for this PR? What changes does it include? Include as much or as little detail as needed.
```
Comment (SmritiSatya) [inline .github/pull_request_template.md:6]: If the PR is related to a specific JIRA or a GitHub issue, include the issue ID in the PR title.

**Review by SmritiSatya — CHANGES_REQUESTED** [review]

**Review by jenniferopal — APPROVED** [review]
lgtm!

**Pull request merged**